precise_bbcode/bbcode/parser.py

```diff
@@ -402,8 +404,19 @@ def _render_textual_content(self, data, replace_specialchars, replace_links, rep
         if replace_links:
             def linkrepl(match):
                 url = match.group(0)
-                href = url if '://' in url else 'http://' + url
-                return '<a href="{0}">{1}</a>'.format(href, url)
+                for xss in bbcode_settings.URL_XSS_FILTER:
+                    if xss in url:
+                        return url
```

Comment:
This logic is duplicated in three places. Could we define it in a shared method to avoid the duplication (eg. in `core/utils`)?

Reply:
On it